spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/NacosDiscoveryProperties.java

```diff
@@ -279,6 +301,205 @@ public void init() throws Exception {
+		private List<String> ignoredInterfaces = new ArrayList();
+
+		@Value("${spring.util.timeout.sec:${SPRING_UTIL_TIMEOUT_SEC:1}}")
+		private int timeoutSeconds = 1;
+
+		@Value("${spring.cloud.inetutils.use-only-site-local-interfaces}")
+		private boolean useOnlySiteLocalInterfaces = false;
+
+		@Value("${spring.cloud.inetutils.preferred-networks}")
+		private List<String> preferredNetworks = new ArrayList();
+
+		public InetUtilsIPV6() {
+			this.executorService = Executors.newSingleThreadExecutor((r) -> {
+				Thread thread = new Thread(r);
+				thread.setName("spring.cloud.alibaba.inetutilsIPV6");
+				thread.setDaemon(true);
+				return thread;
+			});
+		}
+
+		public void close() {
+			this.executorService.shutdown();
+		}
+
+		public InetUtils.HostInfo findFirstNonLoopbackHostInfo() {
+			InetAddress address = this.findFirstNonLoopbackAddress();
+			if (address != null) {
+				return this.convertAddress(address);
+			} else {
+				InetUtils.HostInfo hostInfo = new InetUtils.HostInfo();
+				hostInfo.setHostname(defaultHostname);
+				hostInfo.setIpAddress(defaultIpAddress);
+				return hostInfo;
+			}
+		}
+
+		public InetAddress findFirstNonLoopbackAddress() {
+			InetAddress result = null;
+
+			try {
+				int lowest = 2147483647;
+				Enumeration nics = NetworkInterface.getNetworkInterfaces();
+
+				label61:
+				while(true) {
+					NetworkInterface ifc;
+					do {
+						while(true) {
+							do {
+								if (!nics.hasMoreElements()) {
+									break label61;
+								}
+
+								ifc = (NetworkInterface)nics.nextElement();
+							} while(!ifc.isUp());
+
+							this.log.trace("Testing interface: " + ifc.getDisplayName());
+							if (ifc.getIndex() >= lowest && result != null) {
+								if (result != null) {
+									continue;
+								}
+								break;
+							}
+
+							lowest = ifc.getIndex();
+							break;
+						}
+					} while(this.ignoreInterface(ifc.getDisplayName()));
+
+					Enumeration addrs = ifc.getInetAddresses();
+
+					while(addrs.hasMoreElements()) {
+						InetAddress address = (InetAddress)addrs.nextElement();
+						if (address instanceof Inet6Address && !address.isLoopbackAddress() && this.isPreferredAddress(address)) {
+							this.log.trace("Found non-loopback interface: " + ifc.getDisplayName());
+							result = address;
+						}
+					}
+				}
+			} catch (IOException var8) {
+				this.log.error("Cannot get first non-loopback address", var8);
+			}
+
+			if (result != null) {
+				return result;
+			} else {
+				try {
+					return InetAddress.getLocalHost();
+				} catch (UnknownHostException var7) {
+					this.log.warn("Unable to retrieve localhost");
+					return null;
+				}
+			}
+		}
+
+		 boolean isPreferredAddress(InetAddress address) {
+			 if (this.isUseOnlySiteLocalInterfaces()) {
+				 boolean siteLocalAddress = address.isSiteLocalAddress();
+				 if (!siteLocalAddress) {
+					 this.log.trace("Ignoring address: " + address.getHostAddress());
```

Comment:
Please use log.trace("Ignoring address: {}",address.getHostAddress()); instead of this.log.trace("Ignoring address: " + address.getHostAddress());

Reply:
Required type and provided not suitable